sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py

```diff
@@ -208,18 +226,28 @@ def send_batch(self, event_data_batch, **kwargs):
                 :caption: Sends event data
 
         """
+        partition_id = kwargs.get("partition_id")
+        partition_key = kwargs.get("partition_key")
+        if isinstance(event_data_batch, EventDataBatch):
+            if partition_id or partition_key:
+                raise TypeError("partition_id and partition_key should be None when sending an EventDataBatch "
+                                "because type EventDataBatch itself may have partition_id or partition_key")
+            to_send_batch = event_data_batch
+        else:
+            to_send_batch = self.create_batch(partition_id=partition_id, partition_key=partition_key)
+            to_send_batch._load_events(event_data_batch)  # pylint:disable=protected-access
```

Comment:
Why you make _load_events internal then call an internal method from outside?

Reply:
Because I don't want to expose `_load_events` to users. Users can use class `EventDataBatch`

Reply:
The question is if EventDataBatch.load_events() is publicly accessible. If you don't encourage user to use it, why you use it?

Reply:
As per our offline discussion. EventDataBatch._load_events is good for internal code reuse. But users don't need it.
We can a broad discussion separately whether this way should be used.